Models/ImageClassification/SqueezeNet.swift

```diff
@@ -0,0 +1,93 @@
+// Copyright 2019 The TensorFlow Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import TensorFlow
+
+// Original Paper:
+// SqueezeNet: AlexNet Level Accuracy with 50X Fewer Parameters
+// Forrest N. Iandola, Song Han, Matthew W. Moskewicz, Khalid Ashraf, William J. Dally and Kurt Keutzer
+// https://arxiv.org/pdf/1602.07360.pdf
+
+public struct Fire: Layer {
+
+    public var squeeze: Conv2D<Float>
+    public var expand1: Conv2D<Float>
+    public var expand3: Conv2D<Float>
+
+    public init(inputFilterCount:Int, s1x1: Int, e1x1: Int, e3x3: Int) {
+        squeeze = Conv2D(filterShape: (1, 1, inputFilterCount, s1x1), activation: relu)
+        expand1 = Conv2D(filterShape: (1, 1, s1x1, e1x1), activation: relu)
+        expand3 = Conv2D(filterShape: (3, 3, s1x1, e3x3), padding: .same, activation: relu)
+
+    }
+    @differentiable
+    public func callAsFunction(_ input: Tensor<Float>) -> Tensor<Float> {
+        let squeezed = squeeze(input)
+        let expanded1 = expand1(squeezed)
+        let expanded3 = expand3(squeezed)
+        return expanded1.concatenated(with: expanded3, alongAxis: -1)
```

Comment:
I'd suggest turning the last three statements into a one-liner. 
```suggestion
        return Tensor(concatenating: [expand1(squeezed), expand3(squeezed)], alongAxis: -1)
```

Reply:
That is what I initially tried but it made the function non-differentiable.
`note: cannot differentiate through a non-differentiable argument; do you want to use 'withoutDerivative(at:)'?
        return Tensor(concatenating: [expand1(squeezed), expand3(squeezed)])`